source/NPC.cpp

```diff
@@ -131,6 +137,61 @@ void NPC::Load(const DataNode &node)
 			else
 				location.Load(child);
 		}
+		else if(child.Token(0) == "waypoint")
+		{
+			// Waypoints can be added one-by-one (e.g. if from LocationFilters) or
+			// from multiple "waypoint" nodes. If no system or filter to pick a system is
+			// passed to the node, the mission's destination system will be used.
+			if(!child.HasChildren())
+			{
+				// Given "waypoint" or "waypoint <system 1> .... <system N>"
+				if(child.Size() == 1)
+					missingWaypoint = true;
+				else
+					for(int i = 1; i < child.Size(); ++i)
+						waypoints.push_back(GameData::Systems().Get(child.Token(i)));
+			}
+			// Given "waypoint" and child nodes. These get processed during NPC instantiation.
+			else
+				waypointFilters.emplace_back(child);
+		}
+		else if(child.Token(0) == "destination")
+		{
+			// Only the last defined "destination" node is used to determine the final
+			// destination of the NPC. This destination is visited only after all stopovers
+			// have been, and landing on it completes the mission. If no planet
+			// is passed to the node, the mission's destination will be used.
+			if(!child.HasChildren())
+			{
+				// Given "destination" or "destination <planet>"
+				if(child.Size() == 1)
+					missingDestination = true;
+				else
+					finalDestination = GameData::Planets().Get(child.Token(1));
+			}
+			// Given "destination" and a location filter. These get processed during NPC instantiation.
+			else
+				destinationFilter.Load(child);
+		}
+		else if(child.Token(0) == "stopover")
+		{
+			// Stopovers can be added one-by-one (e.g. if from LocationFilters) or
+			// from multiple "stopover" nodes. All stopovers are visited (no permanent
+			// landing on the stopovers). If no planet is passed to the node, the
+			// mission's destination will be used.
+			if(!child.HasChildren())
+			{
+				// Given "stopover" or "destination <planet 1> ... <planet N>".
```

Comment:
Based on what you said in an earlier review comment.
```suggestion
				// Given "stopover" or "destination <planet 1> ... <planet N>"
```